### PR TITLE
[SHELL32] Don't add folder verbs to non-folder RegItems

### PR DESCRIPTION
**Pull request by whindsaks**
RegItems that pretend to be files should not have Folder verbs

Before/After

### CODE REVIEW

**Review by katahiromz — APPROVED** [review]
LGTM.

File: dll/win32/shell32/folders/CRegFolder.cpp [hunk 1]
```diff
@@ -109,7 +109,13 @@ HRESULT CGuidItemContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
             CoTaskMemFree(pwszCLSID);
         }
     }
-    AddClassKeyToArray(L"Folder", hKeys, &cKeys);
+    UINT folderKey = TRUE;
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CRegFolder.cpp:112]:
```suggestion
    BOOL isFolder = TRUE;
```

File: dll/win32/shell32/folders/CRegFolder.cpp [hunk 2]
```diff
@@ -109,7 +109,13 @@ HRESULT CGuidItemContextMenu_CreateInstance(PCIDLIST_ABSOLUTE pidlFolder,
...
+    SFGAOF att = SFGAO_FOLDER;
+    if (psf && SUCCEEDED(psf->GetAttributesOf(1, apidl, &att)))
+        folderKey = (att & SFGAO_FOLDER);
```
Comment (HBelusca) [inline dll/win32/shell32/folders/CRegFolder.cpp:115]:
```suggestion
        isFolder = !!(att & SFGAO_FOLDER);
```

**Review by HBelusca — COMMENTED** [review]
We should default to folder behaviour if the attributes cannot be retrieved?

**Comment by whindsaks** [comment]
> We should default to folder behaviour if the attributes cannot be retrieved?

I would say the RegItem registration is broken if the attribute is not set.

ROS has been forcing the folder verbs for a while, something might break if we stop but we can give it a shot and fix whatever gets broken.

**Pull request merged**